pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -374,7 +374,8 @@ func inconsistentStatus(set *apps.StatefulSet, status *apps.StatefulSetStatus) b
 // is set to the empty string. status's currentReplicas is set to updateReplicas and its updateReplicas
 // are set to 0.
 func completeRollingUpdate(set *apps.StatefulSet, status *apps.StatefulSetStatus) {
-	if set.Spec.UpdateStrategy.Type == apps.RollingUpdateStatefulSetStrategyType &&
+	if (set.Spec.UpdateStrategy.Type == apps.RollingUpdateStatefulSetStrategyType ||
+		set.Spec.UpdateStrategy.Type == apps.OnDeleteStatefulSetStrategyType) &&
```

Comment:
Should this include a change to https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/apis/apps/types.go#L85? 

Also I believe the actual running pod updates are short cut on onDelete. Is the idea that this will keep rerunning and flip once the required manual updates have been completed?